dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/AbstractServiceConfig.java

```diff
@@ -210,7 +211,7 @@ public ProtocolConfig getProtocol() {
     }
 
     public void setProtocol(ProtocolConfig protocol) {
-        setProtocols(Arrays.asList(protocol));
+        setProtocols(new ArrayList<ProtocolConfig>(Arrays.asList(protocol)));

```

Comment:
`setProtocols(new ArrayList<>(protocol));` seems more concise to me.

Reply:
We don't a constructor to receive one single element for ArrayList.class, do we?

Reply:
Yes, how can I make such a mistake :)